src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -648,7 +652,9 @@ gl_sframe drawing_classifier::perform_inference(data_iterator* data) const {
 
     // Send the inputs to the model.
     std::map<std::string, shared_float_array> results =
-        backend->predict({{"input", result_batch.data_info.drawings}});
+        backend->predict({{"input", result_batch.data_info.drawings},
+                          {"num_samples", shared_float_array::wrap(result_batch.data_info.num_samples)}
+                         });
```

Comment:
It seems strange for me to have `num_samples` be a top-level input to the network. Typically this is captured in the first dimension of the input, or perhaps via a weight tensor (if the batch size is fixed in the backend). If all we support is TF for now, can't we just pass a smaller tensor?

Can follow up in a separate PR. But please leave a TODO: I think having `num_samples` here is ultimately technical debt